File: epfl-hc/hc2013/pb_contestant/art2-SvLO8PDxp7NbY6/artLib2.c
/////////////////////////////
// DO NOT SUBMIT THIS FILE //
/////////////////////////////
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <assert.h>
#include "artLib2.h"

const int CODE_FIRST_COLOR_NOT_ONE = -3;
const int CODE_NO_CONNECTION = -4;
const int CODE_TOO_LONG_DIST = -5;
const int CODE_TOUR_EXCESS = -6;

typedef struct {
	int a, b;
} connection;

static int checkSolution(int n, int m, int* a, int* b, const int* teamColors, const int* teamTour) {
	return 0;
}

int* answer(const int teamColors[], const int teamTour[], int N) {
	int* solution = calloc(1+N+3*N, sizeof(int));
	assert(solution);
	solution[0] = N;
	memcpy(solution+1, teamColors, N*sizeof(int));
	memcpy(solution+N+1, teamTour, 3*N*sizeof(int));
	return solution;
}

int main() {
	int numOfCases;
	int n, m;
	scanf("%d",&numOfCases);
	int tt;
	for (tt = 0; tt < numOfCases; tt++) {
		scanf("%d", &n);
		scanf("%d", &m);
		int* schemeA = calloc(m, sizeof(int));
		assert(schemeA);
		int* schemeB = calloc(m, sizeof(int));
		assert(schemeB);
		int i;
		for (i = 0; i < m; i++) {
			int a, b;
			scanf("%d %d",&a, &b);
			schemeA[i] = a;
			schemeB[i] = b;
		}
		int* teamAnswer = draw_literate_painting(n, m, schemeA, schemeB);
		assert(teamAnswer[0] == n);
		
		int checkResult = checkSolution(n, m, schemeA, schemeB, teamAnswer+1, teamAnswer+1+n);
		free(teamAnswer);
		if (checkResult) {
			printf ("INCORRECT - Test case %d: ", tt+1);
			if (checkResult == CODE_FIRST_COLOR_NOT_ONE)
				printf("color of the first vertex in tour is not 1");
			else if (checkResult == CODE_NO_CONNECTION)
				printf("tour includes nonexistent connection");
			else if (checkResult == CODE_TOO_LONG_DIST)
				printf("distance between some pair of colors is bigger than 3 (tour is illiteral)");
			else if (checkResult == CODE_TOUR_EXCESS)
				printf("tour is too long, should be shorter");
			else if (checkResult > 0)
				printf("didn't find color %d", checkResult % n);
			else
				printf("unknown code error");
			printf("\n");
			exit(0);
		}
		free(schemeA);
		free(schemeB);
	}
	printf("COULD BE CORRECT\n");
	return 0;
}

File: epfl-hc/hc2013/pb_contestant/art2-SvLO8PDxp7NbY6/ArtLib2.java
/////////////////////////////
// DO NOT SUBMIT THIS FILE //
/////////////////////////////

import java.io.InputStreamReader;
import java.io.IOException;
import java.io.BufferedReader;
import java.io.OutputStream;
import java.io.PrintWriter;
import java.util.StringTokenizer;
import java.io.InputStream;
import java.util.HashSet;


public class ArtLib2 {
	private static final int CODE_SIZE_MISMATCH = -1;
	private static final int CODE_NO_TOUR = -2;
	private static final int CODE_FIRST_COLOR_NOT_ONE = -3;
	private static final int CODE_NO_CONNECTION = -4;
	private static final int CODE_TOO_LONG_DIST = -5;
	private static final int CODE_TOUR_EXCESS = -6;
	
	private static int checkSolution(int n, int m, Connexion[] scheme, int[] teamColors, int[] teamTour) {
		return 0;
	}

	public static void main(String args[]) {
		InputStream inputStream = System.in;
		OutputStream outputStream = System.out;
		InputReader in = new InputReader(inputStream);
		PrintWriter out = new PrintWriter(outputStream);
		
		int numOfCases;
		int n, m;
		numOfCases = in.nextInt();
		for (int tt = 0; tt < numOfCases; tt++) {
			n = in.nextInt();
			m = in.nextInt();
			Connexion[] scheme = new Connexion[m];
			for (int i = 0; i < m; i++) {
				int a, b;
				a = in.nextInt();
				b = in.nextInt();
				scheme[i] = new Connexion(a, b);
			}
			
			Pair<int[], int[]> teamAnswer = Art2.draw_literate_painting(n, m, scheme);
			int checkResult = checkSolution(n, m, scheme, teamAnswer.f, teamAnswer.s);
			
			if (checkResult != 0) {
				out.printf ("INCORRECT - Test case %d: ", tt+1);
				if (checkResult == CODE_SIZE_MISMATCH)
					out.printf("size mismatch, received %d, expected %d", teamAnswer.f.length, n);
				else if (checkResult == CODE_NO_TOUR)
					out.printf("received empty tour");
				else if (checkResult == CODE_FIRST_COLOR_NOT_ONE)
					out.printf("color of the first vertex in tour is not 1");
				else if (checkResult == CODE_NO_CONNECTION)
					out.printf("tour includes nonexistent connection");
				else if (checkResult == CODE_TOO_LONG_DIST)
					out.printf("distance between some pair of colors is bigger than 3 (tour is illiteral)");
				else if (checkResult == CODE_TOUR_EXCESS)
					out.printf("tour is too long, should be shorter");
				else if (checkResult > 0)
					out.printf("didn't find color %d", checkResult % n);
				else
					out.printf("unknown code error");
				out.printf("\n");
				out.close();
				System.exit(0);
			}
		}
		out.printf("COULD BE CORRECT\n");
		out.close();
	}
}

class InputReader {
	public BufferedReader reader;
	public StringTokenizer tokenizer;

	public InputReader(InputStream stream) {
		reader = new BufferedReader(new InputStreamReader(stream));
		tokenizer = null;
	}

	public String next() {
		while (tokenizer == null || !tokenizer.hasMoreTokens()) {
			try {
				tokenizer = new StringTokenizer(reader.readLine());
			} catch (IOException e) {
				throw new RuntimeException(e);
			}
		}
		return tokenizer.nextToken();
	}

	public int nextInt() {
		return Integer.parseInt(next());
	}
}

class Connexion {
	public int a;
	public int b;
	
	Connexion(int a_value, int b_value) {
		a = a_value; 
		b = b_value; 
	}
}
class Pair<F, S> {         
	public final F f;
	public final S s;

	public Pair(F f, S s) {         
		this.f = f;
		this.s = s;
	}
}
